File: backend/internal/domain/order.go
package domain

import (
	"context"
)

type Order struct {
	ID              string `json:"id"`
	Canal           string `json:"canal"`
	Cantidad        int    `json:"cantidad"`
	Company         string `json:"company"`
	CP              string `json:"cp"`
	CreatedAt       string `json:"createdAt"`
	DaysToDelivery  string `json:"daysToDelivery"`
	Error           string `json:"error"`
	ErrorMessage    string `json:"errorMessage"`
	FechaCompra     string `json:"fechaCompra"`
	FechaEstimada   string `json:"fechaEstimada"`
	FulfillmentType string `json:"fulfillmentType"`
	IsFlash         bool   `json:"isFlash"`
	IsMarketplace   bool   `json:"isMarketplace"`
	NoPedido        string `json:"noPedido"`
	Plan            string `json:"plan"`
	ProductType     string `json:"productType"`
	SKU             string `json:"sku"`
	StoreSelected   string `json:"storeSelected"`
	TipoPago        string `json:"tipoPago"`
	Edd1            string `json:"edd1"`
	Edd2            string `json:"edd2"`
}

type OrderFilters struct {
	Canal           string `json:"canal"`
	Company         string `json:"company"`
	FulfillmentType string `json:"fulfillment_type"`
	ProductType     string `json:"product_type"`
	Page            int    `json:"page"`
	PageSize        int    `json:"page_size"`
}

type OrderStats struct {
	TotalOrders            int64            `json:"total_orders"`
	BreakdownByCanal       map[string]int64 `json:"breakdown_by_canal"`
	BreakdownByFulfillment map[string]int64 `json:"breakdown_by_fulfillment"`
	BreakdownByProductType map[string]int64 `json:"breakdown_by_product_type"`
	PercentageWithErrors   float64          `json:"percentage_with_errors"`
}

type OrderFiltersOptions struct {
	Channels         []string `json:"channels"`
	Companies        []string `json:"companies"`
	FulfillmentTypes []string `json:"fulfillmentTypes"`
	ProductTypes     []string `json:"productTypes"`
}

type OrderRepository interface {
	List(ctx context.Context, filters OrderFilters) ([]Order, int64, error)
	GetStats(ctx context.Context) (*OrderStats, error)
	GetFilters(ctx context.Context) (*OrderFiltersOptions, error)
}

type OrderService interface {
	ListOrders(ctx context.Context, filters OrderFilters) ([]Order, int64, error)
	GetStats(ctx context.Context) (*OrderStats, error)
	GetFilters(ctx context.Context) (*OrderFiltersOptions, error)
}

File: frontend/src/types/order.ts
// Core order interface — only fields used by the current UI
export interface Order {
  id: string;
  noPedido: string;
  canal: string;
  cantidad: number;
  company: string;
  sku: string;
  fechaCompra: string;
  fechaEstimada: string;
  fulfillmentType: string;
  productType: string;
  createdAt?: string;
  daysToDelivery?: string;
  error?: string;
  errorMessage?: string;
  isFlash?: boolean;
  isMarketplace?: boolean;
  plan?: string;
  // Legacy fields kept for backward-compat with old API responses
  code?: string;
  orderStatus?: OrderStatus;
}

export interface OrderStatus {
  code: string;
  description: string;
  id: number;
  can_cancel: boolean;
}

export interface OrdersResponse {
  data: Order[];
  total: number;
  page: number;
  page_size: number;
  total_pages?: number;
  total_rows?: number;
}

// Normalised shape returned by useOrders (after adapter transform)
export interface NormalisedOrdersPage {
  data: Order[];
  total_rows: number;
  total_pages: number;
  current_page: number;
  page_size: number;
}
